frontend/public/components/utils/k8s-watch-hook.ts

```diff
@@ -0,0 +1,157 @@
+import * as React from 'react';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
```

Comment:
I'm looking into this but hitting an issue related to HOCs, so it's not as simple as bumping `@types/react-redux` version.

https://github.com/DefinitelyTyped/DefinitelyTyped/issues/31363

The `connect` function yields type errors when used with HOCs.

https://github.com/DefinitelyTyped/DefinitelyTyped/issues/31363#issuecomment-460781404